packages/main/src/StepInput.js

```diff
@@ -0,0 +1,651 @@
+ *
+ * App developers can set a maximum and minimum value for the <code>StepInput</code>.
+ * The increase/decrease button and the up/down keyboard navigation become disabled when
+ * the value reaches the max/min or a new value is entered from the input which is greater/less than the max/min.
+ *
+ * <i>When to use</i>
+ * <ul>
+ * <li>To adjust amounts, quantities, or other values quickly.</li>
+ * <li>To adjust values for a specific step.</li>
+ * </ul>
+ *
+ * <i>When not to use</i>
+ * <ul>
+ * <li>To enter a static number (for example, postal code, phone number, or ID). In this case,
+ * use the regular <code>ui5-input</code> instead.</li>
+ * <li>To display a value that rarely needs to be adjusted and does not pertain to a particular step.
+ * In this case, use the regular <code>ui5-input</code> instead.</li>
+ * <li>To enter dates and times. In this case, use date/time related components instead.</li>
+ * </ul>
+ *
+ * For the <code>ui5-step-input</code>
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import @ui5/webcomponents/dist/StepInput.js";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.main.StepInput
+ * @extends UI5Element
+ * @tagname ui5-step-input
+ * @since 1.0.0-rc.12
+ * @public
+ */
+class StepInput extends UI5Element {
+
+	constructor() {
+		super();
+		this.i18nBundle = getI18nBundle("@ui5/webcomponents");
+
+		// spin variables
+		this.INITIAL_WAIT_TIMEOUT = 500; // milliseconds
+		this.ACCELERATION = 0.8;
+		this.MIN_WAIT_TIMEOUT = 50; // milliseconds
+		this.INITIAL_SPEED = 120; // milliseconds
+
+		if (this._previousValue === undefined) {
+			this._previousValue = this.value;
+		}
+	}
+
+	static get metadata() {
+		return metadata;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get styles() {
+		return StepInputCss;
+	}
+
+	static get template() {
+		return StepInputTemplate;
+	}
+
+	static get dependencies() {
+		return [
+			Icon,
+			Input,
+		];
+	}
+
+	static async onDefine() {
+		await fetchI18nBundle("@ui5/webcomponents");
+	}
+
+	get type() {
+		return InputType.Number;
+	}
+
+	// icons-related
+
+	get decIconTitle() {
+		return this.i18nBundle.getText(STEPINPUT_DEC_ICON_TITLE);
+	}
+
+	get decIconName() {
+		return "less";
+	}
+
+	get incIconTitle() {
+		return this.i18nBundle.getText(STEPINPUT_INC_ICON_TITLE);
+	}
+
+	get incIconName() {
+		return "add";
+	}
+
+	get _decIconClickable() {
+		return !this._decIconDisabled && !this.readonly && !this.disabled;
+	}
+
+	get _incIconClickable() {
+		return !this._incIconDisabled && !this.readonly && !this.disabled;
+	}
+
+	get _isFocused() {
+		return this._focused;
+	}
+
+	get _valuePrecisioned() {
+		return this.value.toFixed(this.valuePrecision);
+	}
+
+	get inputAttributes() {
+		return {
+			min: isNaN(this.min) ? undefined : this.min,
+			max: isNaN(this.max) ? undefined : this.max,
+			step: this.step,
+		};
+	}
+
+	onBeforeRendering() {
+		this._buttonsState();
```

Comment:
I would suggest having a verb in the name of each method, f.e. `syncButtonState` or `calculateButtonState` or `setButtonState`

Reply:
Done